src/appleseed-max-impl/logtarget.cpp

```diff
@@ -71,16 +99,29 @@ void LogTarget::write(
         break;
     }
 
-    std::vector<std::string> lines;
-    asf::split(message, "\n", lines);
+    auto is_ui_thread = GetCOREInterface15()->GetMainThreadID() == GetCurrentThreadId();
 
-    for (auto line : lines)
+    if (is_ui_thread)
```

Comment:
What's the logic here? Why do we sometimes write to the log file, sometimes somewhere else?

Reply:
Hi Franz,
I'm yet to fix all your comments, but I wanted to answer this first - so, I had to change this so that it only writes to log from main thread. Otherwise it was crashing during interactive session, since it run from interactive session thread.
I guess during production render this part always runs from main thread.

Reply:
Got it, it's obvious now, I was just confused temporarily :)